### ISSUE TITLE
cylc graph - search for task?

### ISSUE THREAD

**Issue by ColemanTom** [cylc/cylc#2581]
I was thinking it might be useful to be able to, in graph view, do a search for a specific task. In particular, I've done a `cylc graph suite.rc.processed` on a particular system, curious to see the triggering around a specific task (I find this suites graph section a bit large to follow from casual inspection). But then, I had to find the task in the `cylc graph` display. It would be handy to have a navigate to task matching some glob or regex pattern, and you can cycle through the matches.

Is this at all feasible?

I'm not talking about a running suite, as typically they wouldn't display every single task all at once, so are a bit more manageable.

**Comment by matthewrmshin** [comment]
This is sort of related to #2566.

**Comment by hjoliver** [comment]
Yeah, this would be useful, and is feasible (but I think we have higher priorities for a while).

**Comment by hjoliver** [comment]
This could be extended to a static version of #2566: we should able to generate a graph of just the  dependencies around a given task (to a configurable depth).
